folly/experimental/test/StringKeyedTest.cpp

```diff
@@ -129,6 +129,11 @@ using LeakCheckedUnorderedSet = BasicStringKeyedUnorderedSet<
     std::equal_to<folly::StringPiece>,
     MemoryLeakCheckerAllocator<std::allocator<std::string>>>;
 
+#if __CLANG_PREREQ(7, 0)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wself-assign-overloaded"
```

Comment:
Might be able to use pure-C++ tricks to evade the warning.

Reply:
Yep, static cast with a decltype to itself silences the warning. Motivation from https://reviews.llvm.org/D44883

Reply:
@yfeldblum just reworded the commit.  If you would prefer that I put a comment of why we're doing 

```
my_var = static_cast<decltype(my_var)>(my_var);
```

instead of 

```
my_var = my_var;
```

I'm happy to do so.